dialer.ts

```diff
@@ -0,0 +1,65 @@
+import { existsSync } from "fs";
+import { resolve } from "path";
+
+type Dialer = (context: any) => Promise<void> | void;
+
+class DialerRegistry {
+  private preDialers: Dialer[] = [];
+  private postDialers: Dialer[] = [];
+
+  registerPreDialer(dialer: Dialer) {
+    this.preDialers.push(dialer);
+  }
+
+  registerPostDialer(dialer: Dialer) {
+    this.postDialers.push(dialer);
+  }
+
+  async executePreDialers(context: any) {
+    if (this.preDialers.length === 0) {
+      return;
+    }
+
+    for (const dial of this.preDialers) {
+      await dial(context);
+    }
+  }
+
+  async executePostDialers(context: any) {
+    if (this.postDialers.length === 0) {
+      return;
+    }
+
+    for (const dial of this.postDialers) {
+      await dial(context);
+    }
+  }
+}
+
+export const dialerRegistry = new DialerRegistry();
```

Comment:
`DialerRegistry` should take a `path` through the constructor and instantiated in the root, app.ts.